components/org.wso2.carbon.multitenancy.deployment.service/src/main/java/org/wso2/carbon/multitenancy/deployment/service/kubernetes/KubernetesDeploymentProvider.java

```diff
@@ -144,61 +164,149 @@ public Deployment getDeployment(String id) throws DeploymentNotFoundException {
+                if (labels == null) {
+                    labels = new HashMap<>();
+                    kubernetesDeployment.getSpec().getTemplate().getMetadata().setLabels(labels);
+                }
+                setWSO2Labels(labels, deployment, profile);
+
+                logger.debug("Creating deployment: " + kubernetesDeployment.getMetadata().getName());
                 kubernetesClient.extensions().deployments().create(kubernetesDeployment);
+                logger.info("Deployment created: " + kubernetesDeployment.getMetadata().getName());
+                entityFound = true;
             }
             // Add service
             if (resources.containsKey(KIND_SERVICE)) {
                 Service service = new Yaml().loadAs(resources.get(KIND_SERVICE), Service.class);
+                service.getMetadata().setNamespace(namespace);
+                Map<String, String> labels = service.getMetadata().getLabels();
+                if (labels == null) {
+                    labels = new HashMap<>();
+                    service.getMetadata().setLabels(labels);
+                }
+                setWSO2Labels(labels, deployment, profile);
+
+                // Set int value of port if string is set
+                for (ServicePort port : service.getSpec().getPorts()) {
+                    moveStringValToIntVal(port.getTargetPort());
+                }
+
+                logger.debug("Creating service: " + service.getMetadata().getName());
                 kubernetesClient.services().create(service);
+                logger.info("Service created: " + service.getMetadata().getName());
+                entityFound = true;
             }
             // Add ingress
             if (resources.containsKey(KIND_INGRESS)) {
                 Ingress ingress = new Yaml().loadAs(resources.get(KIND_INGRESS), Ingress.class);
+                ingress.getMetadata().setNamespace(namespace);
+                Map<String, String> labels = ingress.getMetadata().getLabels();
+                if (labels == null) {
+                    labels = new HashMap<>();
+                    ingress.getMetadata().setLabels(labels);
+                }
+                setWSO2Labels(labels, deployment, profile);
+
+                // Set int value of port if string is set
+                if (ingress.getSpec().getRules() != null) {
+                    for (IngressRule rule : ingress.getSpec().getRules()) {
+                        if (rule.getHttp() != null) {
+                            for (HTTPIngressPath path : rule.getHttp().getPaths()) {
+                                if (path.getBackend() != null) {
+                                    moveStringValToIntVal(path.getBackend().getServicePort());
+                                }
+                            }
+                        }
+                    }
+                }
+
+                logger.debug("Creating ingress: " + ingress.getMetadata().getName());
```

Comment:
debug log check is needed as this is;
1. inside a loop 
2. A method call and string contact is happening all the time